fuzz/href.cc

```diff
@@ -0,0 +1,35 @@
+#define ADA_DEVELOPMENT_CHECKS 1
+
+#include <fuzzer/FuzzedDataProvider.h>
+
+#include <memory>
+#include <string>
+
+#include "ada.cpp"
+#include "ada.h"
+
+extern "C" int LLVMFuzzerTestOneInput(const uint8_t *data, size_t size) {
+  FuzzedDataProvider fdp(data, size);
+  std::string source = fdp.ConsumeRandomLengthString(256);
+  std::string base_source = fdp.ConsumeRandomLengthString(256);
+
+  auto out_url = ada::parse<ada::url>(source);
```

Comment:
This fuzzer runs super similar to `parse`. I've opened a PR to improve it: https://github.com/ada-url/ada/pull/665

Reply:
Sounds good